controller/src/main/java/org/jboss/as/controller/registry/ImmutableManagementResourceRegistration.java

```diff
@@ -252,4 +253,11 @@
      * @return the set of ordered child types, and and empty set if there are none
      */
     Set<String> getOrderedChildTypes();
+
+    /**
+     * Returns capability definition for this resource.
+     *
+     * @return RuntimeCapability if any registered otherwise null
+     */
+    Set<RuntimeCapability<?>> getCapabilities();
```

Comment:
No, that's not a problem as that's not a leak of the instance to someone other than the provider or the kernel. It passes the instance from provider to kernel. What I wanted to avoid was other subsystems seeing an API call to get a ref to someone else's RuntimeCapability and deciding it was appropriate to use that.